sdks/python/apache_beam/runners/interactive/utils.py

```diff
@@ -267,3 +270,17 @@ def return_as_json(*args, **kwargs):
       return str(return_value)
 
   return return_as_json
+
+
+def deferred_df_to_pcollection(df):
+  assert isinstance(df, DeferredBase), '{} is not a DeferredBase'.format(df)
+
+  # The proxy is used to output a DataFrame with the correct columns.
+  #
+  # TODO(BEAM-11064): Once type hints are implemented for pandas, use those
+  # instead of the proxy.
+  cache = ExpressionCache()
+  cache.replace_with_cached(df._expr)
+
+  proxy = df._expr.proxy()
+  return to_pcollection(df, yield_elements='pandas', label=str(df._expr)), proxy
```

Comment:
Is the proxy basically the `pcoll.element_type`? Will this be redundant?

Reply:
No not yet, the proxy is different than the pcoll.element_type. The proxy is a tool used in the Beam DataFrames implementation to keep track of the column names and index. Once the TODO above is fixed, however, then it can be replaced with the pcoll.element_type.